src/kubernetes_cluster/spec/external_api.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use crate::external_api::spec::*;
+use crate::kubernetes_cluster::spec::message::*;
+use crate::state_machine::action::*;
+use crate::state_machine::state_machine::*;
+use crate::temporal_logic::defs::*;
+use vstd::{multiset::*, prelude::*};
+
+verus! {
+
+// This enum is a wrapper for attaching rest id to the input and output of the external api.
+// By doing this, we are able to check whether the output matches the input when conducting continue_reconcile.
+#[is_variant]
+pub enum ExternalComm<Input, Output> {
```

Comment:
This is a crucial data type so we should document it carefully. Could you revise the comment in this way:
ExternalComm is used to represent xxx. It has two variants: Input is xxx and Output is xxx. Each variant also carries a nat which represents xxx.

btw we should not call the nat here as "rest id" -- ExternalComm is not necessarily REST API call. We can just call these nat as "call_id".

Reply:
And, could you also document this data type briefly in the description?

Reply:
>  ExternalComm is not necessarily REST API call. We can just call these nat as "call_id".

I can call it call id in the document. But it is also generated by rest_id of the state, right? Or do you think we should create another id?

Reply:
I see. It should be generated by rest_id. Maybe I should rename the top level rest_id to sth else